Source/Core/Core/PowerPC/JitInterface.cpp

```diff
@@ -123,58 +107,97 @@ void JitInterface::UpdateMembase()
   }
 }
 
-void JitInterface::WriteProfileResults(const std::string& filename) const
+static std::string_view GetDescription(const CPUEmuFeatureFlags flags)
 {
-  Profiler::ProfileStats prof_stats;
-  GetProfileResults(&prof_stats);
-
-  File::IOFile f(filename, "w");
-  if (!f)
-  {
-    PanicAlertFmt("Failed to open {}", filename);
-    return;
-  }
-  f.WriteString("origAddr\tblkName\trunCount\tcost\ttimeCost\tpercent\ttimePercent\tOvAllinBlkTime("
-                "ms)\tblkCodeSize\n");
-  for (auto& stat : prof_stats.block_stats)
-  {
-    std::string name = m_system.GetPPCSymbolDB().GetDescription(stat.addr);
-    double percent = 100.0 * (double)stat.cost / (double)prof_stats.cost_sum;
-    double timePercent = 100.0 * (double)stat.tick_counter / (double)prof_stats.timecost_sum;
-    f.WriteString(fmt::format("{0:08x}\t{1}\t{2}\t{3}\t{4}\t{5:.2f}\t{6:.2f}\t{7:.2f}\t{8}\n",
-                              stat.addr, name, stat.run_count, stat.cost, stat.tick_counter,
-                              percent, timePercent,
-                              static_cast<double>(stat.tick_counter) * 1000.0 /
-                                  static_cast<double>(prof_stats.countsPerSec),
-                              stat.block_size));
-  }
+  static constexpr std::array<std::string_view, (CPUEmuFeatureFlags::EndOfEnumeration - 1) << 1>
+      descriptions = {
+          "", "DR", "IR", "DR|IR", "PERFMON", "DR|PERFMON", "IR|PERFMON", "DR|IR|PERFMON",
+      };
+  return descriptions[flags];
 }
 
-void JitInterface::GetProfileResults(Profiler::ProfileStats* prof_stats) const
+void JitInterface::JitBlockLogDump(const Core::CPUThreadGuard& guard, std::FILE* file) const
 {
-  // Can't really do this with no m_jit core available
+  std::fputs("ppcFeatureFlags\tppcAddress\tppcSize\tppcCycles\thostNearSize\thostFarSize\trunCount"
+             "\tcyclesSpent\tcyclesPercent\ttimeSpent(μs)\ttimeAverage(ns)\ttimePercent\tsymbol\n",
+             file);
+
   if (!m_jit)
     return;
 
-  prof_stats->cost_sum = 0;
-  prof_stats->timecost_sum = 0;
-  prof_stats->block_stats.clear();
-
-  const Core::CPUThreadGuard guard(m_system);
-  QueryPerformanceFrequency((LARGE_INTEGER*)&prof_stats->countsPerSec);
-  m_jit->GetBlockCache()->RunOnBlocks([&prof_stats](const JitBlock& block) {
-    const auto& data = block.profile_data;
-    u64 cost = data.downcountCounter;
-    u64 timecost = data.ticCounter;
-    // Todo: tweak.
-    if (data.runCount >= 1)
-      prof_stats->block_stats.emplace_back(block.effectiveAddress, cost, timecost, data.runCount,
-                                           block.codeSize);
-    prof_stats->cost_sum += cost;
-    prof_stats->timecost_sum += timecost;
-  });
-
-  sort(prof_stats->block_stats.begin(), prof_stats->block_stats.end());
+  if (std::strcmp(m_jit->GetName(), "Cached Interpreter") == 0)
```

Comment:
This is fragile in case someone changes the name of the cached interpreter. If you must do it using string comparison instead of using an enum, please define the name of the cached interpreter in a header file somewhere, and use that definition both here and in the GetName implementation.

Reply:
Was able to remove this special path by following your suggestion about the near/far begin/end pointers. The profile data is still all zero-initialized for the Cached Interpreter, but we'll just consider it a TODO and leave it at that.